contributors/design-proposals/add-metadata-indicating-to-replace-union-in-patch.md

```diff
@@ -0,0 +1,280 @@
+Add Tag Indicating To Replace *Union* Fields in Patch (a.k.a Support Replacing Maps in Strategic Merge Patch)
+=============
+
+This approach could be used to fix the issues with `kubectl apply` simply by having apply always replace union fields instead of patching them.  E.g. instead of setting `spec.strategy.type` without touching the rest of `spec.strategy`, patches should always replace the entire `spec.strategy`.
+
+## Proposed Changes
+
+### APIs
+
+Add a go struct tag/openAPI value telling clients to replace the entire struct when doing a *PATCH*. For the *inline* fields, we have 2 options:
+
+1) Move the tag to the parent struct's. E.g. add a tag to `Volume` instead of `VolumeSource`. This will limit the flexibility to add more fields in the future.
+
+2) Add some logic when lookup the metadata in go struct to find the tags for inline fields.
+
+Examples of how to add go struct tag respect option 2:
+```go
+type Volume struct {
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	VolumeSource `json:",inline" protobuf:"bytes,2,opt,name=volumeSource" patchStrategy:"replace"`
+}
+
+type DeploymentSpec struct {
+	...
+	Strategy DeploymentStrategy `json:"strategy,omitempty" protobuf:"bytes,4,opt,name=strategy" patchStrategy:"replace"`
+	...
+}
+```
+
+### kubectl
+
+No required change.
+
+## Strategic Merge Patch
+
+Add logic to support replacing the entire union fields when there is a `replace` directive. The logic will be similar to the logic of merge strategy for lists. 
```

Comment:
How would this be implemented?  Does the server know to replace the entire object?  Or does the client explicitly clear these fields?